org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/DiagnosticResolution.java

```diff
@@ -121,7 +119,7 @@ public WorkspaceEdit apply() {
 			EObject obj = helper.resolveContainedElementAt(tmpResource, offset);
 
 			WorkspaceEdit edit = new WorkspaceEdit();
-			Diagnostic diagnostic = params.getContext().getDiagnostics().stream().collect(MoreCollectors.onlyElement());
+			Diagnostic diagnostic = params.getContext().getDiagnostics().stream().findFirst().get();
```

Comment:
wont this throw if there is no diagnostic? should we catch?

Reply:
Yes, it will (same as MoreCollectors.onlyElement()). It fill be cached by the general handler some lines below. If you like I can handle it in a nicer way, even though I am not sure if it is possible to get that far without diagnostics.

Reply:
ok

Reply:
I would indeed handle it a bit nicer. I would also place it a bit higher in the method for an early exit

```java
try {
List<Diagnostic> diagnostics = params.getContext().getDiagnostics();
if (diagnostics.isEmpty()) {
    log.warn("No diagnostics present for code action")
    return null;
}

Diagnostic diagnostic = Iterables.getFirst(params.getContext().getDiagnostics())
```

Reply:
other question: do we need to pick / filter the "correct" diagnostic. am not sure how the diagostics are sorted and we the quickfix can handle a incorrect one.

Reply:
We should not need to pick the "correct" diagnostic, that happened already before. It looks to me we just need to pick the diagnostic we chose already. 

I have pushed a new version of the PR to to that and have reorganized the DiagnosticResolution.apply() method a little bit to declare variables in the order they are used.